Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -225,9 +230,43 @@ public void run(@NotNull ProgressIndicator indicator) {
     @Override
     public void beforeDocumentChange(DocumentEvent e) {
       myServerData.onDocumentChanged(e);
+      synchronized (myLock) {
+        final Document document = e.getDocument();
+        myUpdatedDocuments.add(document);
+        scheduleUpdateFilesContent();
+      }
     }
   };
 
+  private void scheduleUpdateFilesContent() {
```

Comment:
Maybe a little theoretical problem, but...
Will it work if some background process will update the document of an invisible editor?
So, we would not need to highlight it, but the change would cause creating new errors, changes in index, etc.